h2/src/main/org/h2/util/DateTimeUtils.java

```diff
@@ -902,4 +902,17 @@ public static long dateValueFromAbsoluteDay(long absoluteDay) {
         return dateValue(y, m + 3, (int) d);
     }
 
+    /**
+     * Adds the number of months to the date. If the resulting month's number of days is less than the original's day-of-month, the resulting
+     * day-of-months gets adjusted accordingly:
+     *
+     * 30.04.2007 - 2 months = 28.02.2007
+     */
+    public static Timestamp addMonths(final Timestamp refDate, final int nrOfMonthsToAdd) {
+        Calendar calendar = Calendar.getInstance();
+        calendar.setTime(refDate);
+        calendar.add(Calendar.MONTH, nrOfMonthsToAdd);
+        return new Timestamp(calendar.getTime().getTime());
```

Comment:
2 things
(1) this is going to drop the nanosecond component, you will need to copy that over as well 
(2) rather use calendar.getTimeInMillis(), one less intermediate object